### DISCUSSION TITLE
[Feature] Viewing and Indexing Local Photos in Offline Mode

### DISCUSSION THREAD

**Discussion by OriginalSimon** [immich-app/immich#14694] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

**Issue Description:**  
Add offline mode functionality to the Immich app, allowing users to view photos stored locally on the device but not yet uploaded to the server.

This is particularly important for situations where the server is unavailable (e.g., no internet connection), so that the app loads instantly, indexing and displaying local photos. I want to make Immich my primary photo viewing app, including offline access.

Currently, this is not possible because, after disconnecting from the home server, new photos taken after leaving home are not visible until the device reconnects to the server.

**Example Implementation:**  
Upon startup, the app scans local photos and displays them, even without a server connection. This behavior is similar to Google Photos, where users can always see their local photos, regardless of server connectivity.

**Required Functionality:**  
1. Instant app launch in offline mode.  
2. Indexing and displaying all local photos.  
3. Marking unsynchronized photos.

### Platform

- [ ] Server
- [ ] Web
- [X] Mobile

**Comment by bo0tzz** [comment]
#1605
